src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact {
+
+    protected String value;
+    protected boolean isPrivate;
+    
+    public Contact() { }
+    
+    public Contact(String value, boolean isPrivate) {
+        this.value = value;
+        this.isPrivate = isPrivate;
+    }
+
```

Comment:
This constructor can be used by the child classes to assign the variables instead.